module-system/module-implementations/sov-bank/src/token.rs

```diff
@@ -62,6 +66,43 @@ impl<C: sov_modules_api::Context> Token<C> {
         Ok(CallResponse::default())
     }
 
+    pub(crate) fn freeze(&mut self, sender: &C::Address) -> Result<CallResponse> {
+        self.is_authorized_minter(sender)?;
+        if self.frozen {
+            bail!("Token is already frozen")
+        }
+        self.frozen = true;
+        Ok(CallResponse::default())
```

Comment:
Nit: Can this function just return `()`,  It looks like this function don't have to be aware of this type.